src/systems/remove-object3D-system.js

```diff
@@ -62,6 +63,14 @@ const cleanupSkyboxes = cleanupObjOnExit(Skybox, obj => {
   disposeMaterial(obj.sky.material);
   obj.sky.geometry.dispose();
 });
+const cleanupSimpleWaters = cleanupObjOnExit(SimpleWater, obj => {
+  obj.geometry.dispose();
+  if (Array.isArray(obj.material)) {
+    obj.material.forEach(material => disposeMaterial(material));
+  } else {
+    disposeMaterial(obj.material);
+  }
```

Comment:
You may be able to replace this cleanup with `disposeNode`: https://github.com/mozilla/hubs/blob/eb5da30aa145133ce822d29928978850a75c0b1c/src/utils/three-utils.js#L36-L50

I'm not sure if a `SimpleWater` is a `THREE.Mesh`, but maybe `disposeNode` can handle that if necessary.

Reply:
Oops thanks for pointing that, I just realized that I we are using a shared normal map for the water mesh so we should only dispose geo and material but not the textures and both `disposeMaterial` and `disposeNode` do that.

Also I updated the AFrame component to free resources upon removal, we were not doing that.